semantic/analyze_libflux.go

```diff
@@ -16,9 +16,10 @@ func AnalyzeSource(fluxSrc string) (*Package, error) {
 		return nil, err
 	}
 	defer sem.Free()
-	fb, err := sem.MarshalFB()
+	fb, offset, freeFn, err := sem.MarshalFB()
 	if err != nil {
 		return nil, err
 	}
-	return DeserializeFromFlatBuffer(fb)
+	defer freeFn()
```

Comment:
On the current master, this will work fine.  On the `feat/use-algo-w` branch, when deserialize the semantic graph, it will contain a pointer into the flatbuffer for each expression and assignment statement.  We can't free the flatbuffer here since the graph that we return to the caller has pointers into it.  I think we will be stuck with leaving it to the caller to free the memory when it's done.  Curious to know your thoughts on this.

Reply:
Can you explain why do we need to keep so many references to the flatbuffer at different offsets? Why do those expressions still need to refer to the original flatbuffer?

Reply:
Yes, that's the idea.  The semantic graph will be deserialized but it will contain `MonoType` and `PolyType` structs that are thin wrappers around bytes in the original flatbuffer.  There is no Go-native deserialized format for types, i.e., Flux runtime execution will talk directly to the flatbuffers representation of types (via the thin wrapper).

Only `NativeVariableAssignments` have polytypes, and expressions have monotypes, e.g.:
https://github.com/influxdata/flux/blob/feat/use-algo-w/semantic/graph.go#L636

The PolyType and MonoType structs are defined here, they are the wrappers around the Go code generated by `flatc`:
https://github.com/influxdata/flux/blob/feat/use-algo-w/semantic/monotype.go#L14-L24
https://github.com/influxdata/flux/blob/feat/use-algo-w/semantic/polytype.go#L13-L17

I'm seeing now how this generates issues since the lifetime of the byte array needs to be explicitly managed in Go.  I'd be happy to pair on this if that would be helpful.